woocommerce/woocommerce-android
# [Blaze] Payment methods list

Part of: #10721 


### Description
This PR adds a screen that list payment methods list for the Blaze campaign creation flow.

### Testing instructions
1. Sign in using WordPress.com
2. Start Blaze campaign creation.
3. Click on "Confirm details".
4. Click on the Payment Method row.
5. Confirm it opens a screen where the available payment methods are listed (for now the API is mocked, so the list contains two sample credit cards).
6. Select a different payment method.
7. Confirm this navigates you back to the previous screen, and the selected method has been updated.

### Images/gif
| Empty payment methods list | Non-empty payment methods list |
| ---- | ---- |
| ![Screenshot_20240212_120435](https://github.com/woocommerce/woocommerce-android/assets/1657201/27d42116-4fe4-48b5-b48a-1f46025a5019) | ![Screenshot_20240212_115521](https://github.com/woocommerce/woocommerce-android/assets/1657201/64f64481-5f7a-4f31-918c-0b66c1923189) |
